src/shadowbox/server/server_access_key.ts

```diff
@@ -68,6 +70,7 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
   constructor(
       private portProvider: PortProvider, private proxyHostname: string,
       private keyConfig: JsonConfig<AccessKeyConfigJson>,
+      private serverConfig: JsonConfig<ServerConfigJson>,
```

Comment:
Did you consider simply passing the default port number? That'll be easier to mock, for one thing.

Reply:
Done. I now pass to `accessKeyRepository.enableSinglePort()`